File: leetcode/217_ContainsDuplicate/kotlin/ContainsDuplicate.kt
class Solution {
    fun containsDuplicate(nums: IntArray): Boolean {
        if (nums.size < 2) {
            return false
        }

        val map = mutableMapOf<Int, Int>()
        nums.forEach { num ->
            if (map[num] == null) {
                map.put(num, num)
            } else {
                return true
            }   
        }
        return false
    }
}

File: leetcode/217_ContainsDuplicate/typescript/contains-duplicate.ts
function containsDuplicate(nums: number[]): boolean {
    if (nums.length < 2) {
        return false;
    }

    const map = new Map();

    for (let i=0; i<nums.length; i++)   {
        if (map.has(nums[i])) {
            return true;
        } else {
            map.set(nums[i], i);
        }
    }
    return false;
};
